lib/sycamore/sycamore/transforms/summarize.py

```diff
@@ -143,7 +141,238 @@ def collapse(text: str, tokens_per_chunk: int, tokenizer: Tokenizer, summarizer_
     return cur_summary
 
 
-class DocumentSummarizer(Summarizer):
+class HeirarchicalDocumentSummarizer(Summarizer):
+    """
+    Summarizes a document by constructing a heirarchical tree of batches of elements,
```

Comment:
I'm not sure we want this construction; I was expecting something that grouped until it ran out of context window. Do we have  a reason to believe that a multi-stage summarize is better than a single stage one?